oauthlib/oauth2/rfc6749/grant_types/implicit.py

```diff
@@ -328,6 +317,16 @@ def validate_token_request(self, request):
 
         # Then check for normal errors.
 
+        request_info = {}
+        # For implicit grant, auth_validators and token_validators are
+        # basically equivalent since the token is returned from the
+        # authorization endpoint.
+        for validator in chain(self._token_validators_run_before_standard_ones,
```

Comment:
Extract to a helper method since this code is repeated on line 372 with different validators.